### ISSUE TITLE
Enhancement: Warn for globbing inside double quotes?

### ISSUE THREAD

**Issue by jmrgibson** [koalaman/shellcheck#1370]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I realize that this is covered in the wiki, but I wonder what peoples thoughts are about warning the user when they nullify glob behaviour with quotes.

#### Here's a snippet or screenshot that shows the problem:
This has happened to me multiple times now 

```sh

#!/bin/bash
// correctly warns about no quotes
cp ${FOO}/bar.* some_folder

// try to fix, but now behaviour changed with no warning
cp "${FOO}/bar.*" some_folder

// whereis what I should actually type is 
cp "${FOO}"/bar.* some_folder
```

#### Here's what shellcheck currently says:
nothing!

#### Here's what I wanted or expected to see:
Warning: Globbing will not occur inside double quotes
